cupy/random/_generator_api.pyx

```diff
@@ -285,6 +288,49 @@ class Generator:
 
         return y
 
+    def gamma(self, shape, scale=1.0, size=None):
+        """Returns an array of samples drawn from a gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.gamma` for full documentation
+            - :meth:`numpy.random.RandomState.gamma`
+        """
+        return self.standard_gamma(shape, size) * scale
+
+    def standard_gamma(self, shape, size=None, dtype=numpy.float64, out=None):
+        """Returns an array of samples drawn from a standard gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.standard_gamma` for full documentation
+            - :meth:`numpy.random.RandomState.standard_gamma`
+        """
+        cdef ndarray y
+
+        if shape < 0.0:
+            raise ValueError('shape < 0')
```

Comment:
Will the method support `shape: Union[float, cupy.ndarray]`?

Reply:
This is a good question, I think it should not support it because it will be enforcing device synchronization.

Reply:
I have double checked and it also acceps an array of floats, so I have to think about this check.

Reply:
I found `cupy.random.default_rng(0).standard_gamma(-1e-15)` hangs.

How about adding
```
    } else if (shape <= 0.0) {
        return 0.0
```
to `rk_standard_gamma`?

Reply:
sgtm!